cmd/ci-operator/main.go

```diff
@@ -1607,21 +1614,20 @@ func getHashFromBytes(b []byte) string {
 	return oneWayNameEncoding.EncodeToString(hash.Sum(nil)[:5])
 }
 
-func getPullSecretFromFile(filename string) (*coreapi.Secret, error) {
-	secret := &coreapi.Secret{
-		Data: make(map[string][]byte),
-		ObjectMeta: meta.ObjectMeta{
-			Name: steps.PullSecretName,
-		},
-		Type: coreapi.SecretTypeDockerConfigJson,
-	}
-
+func getSecret(name, filename string) (*coreapi.Secret, error) {
 	src, err := ioutil.ReadFile(filename)
 	if err != nil {
-		return nil, fmt.Errorf("could not read file %s for secret: %w", filename, err)
+		return nil, fmt.Errorf("could not read file %s for secret %s: %w", filename, name, err)
```

Comment:
We need to make sure the returned secret is nil for all error cases.
Because later on we will use it as condition to do oc-image-mirror.